pkg/workceptor/pod_test.go

```diff
@@ -0,0 +1,390 @@
+package workceptor_test
+
+import (
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var podSuccess = &corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: "default",
+		Name:      "test-pod",
+	},
+	Status: corev1.PodStatus{
+		Phase: corev1.PodSucceeded,
+		ContainerStatuses: []corev1.ContainerStatus{
+			{
+				Name: "worker",
+				State: corev1.ContainerState{
+					Waiting: nil,
+					Running: nil,
+					Terminated: &corev1.ContainerStateTerminated{
+						ExitCode: 0,
+						Reason:   "Success",
+					},
+				},
+			},
+		},
+	},
+}
+
+var podInfraError = &corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: "default",
+		Name:      "infra-error-pod",
+	},
+	Status: corev1.PodStatus{
+		Phase:  corev1.PodFailed,
+		Reason: "Pod OOMKilled",
+		ContainerStatuses: []corev1.ContainerStatus{
+			{
+				Name: "worker",
+				State: corev1.ContainerState{
+					Terminated: &corev1.ContainerStateTerminated{
+						ExitCode: 137,
+						Reason:   "Container OOMKill",
```

Comment:
```suggestion
		Reason: "OOMKilled",
		ContainerStatuses: []corev1.ContainerStatus{
			{
				Name: "worker",
				State: corev1.ContainerState{
					Terminated: &corev1.ContainerStateTerminated{
						ExitCode: 137,
						Reason:   "OOMKill",
```
This object matches the actual K8s objects that are returned. While I understand the want to change these it would be confusing in a real debugging scenario. K8s does not include Pod/Container in the returned reason so we should not expect it.